modules/openthread/platform/diag.c

```diff
@@ -68,3 +75,159 @@ void otPlatDiagAlarmCallback(otInstance *aInstance)
 {
 	ARG_UNUSED(aInstance);
 }
+
+/*
+ * To enable gpio diag commands, in Devicetree create `gpio_ports` node in root path
+ * with `compatible = "gpio-ports"` property and `gpio-ports-phandles` property,
+ * which should contain phandles of GPIO port nodes. e.g:
+ *
+ * gpio-ports {
+ *		compatible = "gpio-ports";
+ *		gpio-ports-phandles = <&gpio0 &gpio1>;
+ * };
+ *
+ * To enable reading current gpio pin mode, define
+ * `CONFIG_GPIO_GET_DIRECTION` in prj.conf.
+ *
+ * Note: `<gpio>` in `diag gpio` commands is an integer that combines
+ * port and pin into a single, contiguous number space as follows:
+ * gpio = (port * 32) + pin
+ */
+#if DT_NODE_EXISTS(DT_PATH(gpio_ports)) && \
+	DT_NODE_HAS_PROP(DT_PATH(gpio_ports), gpio_ports_phandles)
```

Comment:
last check is redundant, since DT prop is flagged as required, so build will not even start if it's missing.